client/app/visualizations/pivot/index.js

```diff
@@ -21,7 +22,19 @@ function pivotTableRenderer() {
         if ($scope.queryResult.getData() !== null) {
           // We need to give the pivot table its own copy of the data, because it changes
           // it which interferes with other visualizations.
-          data = angular.copy($scope.queryResult.getRawData());
+          data = [];
+          const columns = $scope.queryResult.getColumns();
+          const rows = $scope.queryResult.getData();
+          rows.forEach((row) => {
+            const rowObj = {};
+            columns.forEach((col) => {
+              const colName = getColumnCleanName(col.name);
+              const value = formatValue($filter, clientConfig, row[col.name], col.type);
```

Comment:
- Comment 2

  I am thinking of adding a Sorting option to pivottable editor. Because passing a sort-friendly format will work for numerical dates, but it doesn't help for Monday/Tuesday/..., Mon/Tue/..., January/February/..., Jan/Feb/..., etc.
 If we pass `REDAH_DATE_FORMAT` to pivot, the problem below will go away because the dates won't be in ISO format,

![](https://cloud.githubusercontent.com/assets/5517304/22537206/351ce520-e954-11e6-9bdb-7f322beb066a.png)

and letting people define custom sorters will take care of the first comment of #216.

- Comment 1
I have to look into it a bit more for part 1 of your comment.